user/tests/wiring/no_fixture_power_saving/no_fixture_power_saving.spec.js

```diff
@@ -0,0 +1,128 @@
+suite('No fixture power saving');
+
+platform('boron','bsom','esomx');
+timeout(32 * 60 * 1000);
+
+
+let api = null;
+let auth = null;
+let device = null;
+let deviceId = null;
+let limits = null;
+let skipTest = false;
+
+async function delayMs(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function getDeviceFunctionsWithRetries({ deviceId, auth, expectedFuncNum = 0, retries = 10, delay = 1000 } = {}) {
+    let lastError;
+    for (let i = 0; i < retries; i++) {
+        try {
+            const resp = await api.getDevice({ deviceId, auth });
+            const funcs = resp.body.functions;
+            if (expectedFuncNum > 0 && funcs.length !== expectedFuncNum) {
+                throw new Error('Number of functions returned from device does not match expected');
+            }
+            return resp;
+        } catch (e) {
+            lastError = e;
+        }
+        await delayMs(i * delay);
+    }
+    if (lastError) {
+        // console.log(lastError);
+        throw lastError;
+    }
+    throw new Error('Error fetching functions from device');
+}
+
+before(function() {
+    api = this.particle.apiClient.instance;
+    auth = this.particle.apiClient.token;
+    device = this.particle.devices[0];
+    deviceId = device.id;
+
+    // This code will run after the appropriate mailbox message is received.
+    device.on('mailbox', async (msg) => {
+        if (msg.d === "skip_test") {
+            skipTest = true;
+            return;
+        }
+        // console.log('waiting for device to enter low power');
+        await delayMs(30000);
+        // console.log('waking up device with a function call');
+        let lastError;
+        try {
+            let resp = await api.callFunction({ deviceId, name: 'fnlp1', argument: 'argument string low power sleep', auth });
+            expect(resp.body.return_value).to.equal(12345);
+        } catch (e) {
+            lastError = e;
+        }
+        if (lastError) {
+            // console.log(lastError);
+            throw lastError;
+        }
+    });
+});
+
+test('POWER_SAVING_00_setup', async function() {
+    // See power_saving_mode.cpp
+});
+
+test('POWER_SAVING_01_particle_publish_publishes_an_event_after_low_power_active', async function() {
+    if (skipTest) {
+        return;
+    }
+
+    const data = await this.particle.receiveEvent('my_event_low_power');
+    expect(data).to.equal('event data low power');
+});
+
+test('POWER_SAVING_02_register_function_and_connect_to_cloud', async function() {
```

Comment:
What is this test testing? It looks like the device connects and stays in low power mode, then the test runner side queries it for its device functions? 
This is to ensure that the cloud side can communicate with the device even when the modem enters low power mode?

Reply:
This sets up the device to have a Particle function, and ensures the device is in low power mode.  Then the JS side runs and checks that the functions are registered with the cloud (describe sent).  The real test of low power mode with functions is then the next test case, when the JS side sends a Function call.